### DISCUSSION TITLE
using nunjucks to parse yaml front matter item without knowing its structure

### DISCUSSION THREAD

**Discussion by btrem** [11ty/eleventy#2976] [Q&A] [answered]
I have a yaml front matter item that might be a list:

```yaml
foo:
  - bar1
  - bar2
  - bar3
```
or it might be a dictionary:
```yaml
foo:
  bar1: baz
  bar2: 2
  bar3: 3 baz
```
Is there a way to parse `foo` in a nunjucks template? For the list, I need `{% for item in foo  %}`; for the dictionary, I need `{% for item, value in foo  %}`. But I can't figure if I can determine the nature of foo in nunjucks in order to choose the right `for` loop.

**Comment by pdehaan** [comment, marked as answer]
I couldn't easily figure out how to call `Array.isArray()`, but I was able to do it with a custom filter and a slightly verbose `{% if %}...{% else %}...{% endif %}` block:

```js
eleventyConfig.addFilter("isArray", Array.isArray);
```

### USAGE
```njk
---
title: Bleep bloop
foo2:
  - bar1
  - bar2
  - bar3

foo:
  bar1: baz
  bar2: 2
  bar3: 3 baz
---

{% if foo | isArray %}
  {% for item in foo %}
    - {{ item }}
  {% endfor %}
{% else %}
  {% for item, value in foo %}
    - {{ item }}; {{ value }}
  {% endfor %}
{% endif %}
```
